File: Java/2000_Reverse_Prefix_of_Word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int index = word.indexOf(ch);
        if(index != -1){
            StringBuilder result = new StringBuilder(word.substring(0,index+1));
            result.reverse();
            result.append(word.substring(index+1, word.length()));
            return result.toString();
        }
        return word;
        
        /*
        StringBuilder result = new StringBuilder();
        int i = 0;
        
        for(; i < word.length(); i++){
            result.insert(0, word.charAt(i));// = word.charAt(i) + result;
            if(word.charAt(i) == ch){
                break;
            }
        }
        if(i < word.length()){
            for(int j = i+1; j < word.length(); j++){
                result.append(word.charAt(j));// += word.charAt(j);
            }
        }
        else{
            result = new StringBuilder(word);
        }
        
        return result.toString();
        */
    }
}

File: CS/2000_Reverse_Prefix_of_Word.cs
public class Solution {
    public string ReversePrefix(string word, char ch) {
        StringBuilder result = new StringBuilder();
        int index = word.IndexOf(ch);
        for(int i = index; i >= 0; i--){
            result.Append(word[i]);
        }
        for(int i = index+1; i < word.Length; i++){
            result.Append(word[i]);
        }
        
        return result.ToString();
    }
}
